pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -1004,10 +1041,93 @@ func (service *SecurityPolicyService) createOrUpdateNSXSecurityPolicy(nsxSecurit
 		return nil, err
 	}
 
+	// Check SecurityPolicy realization state
+	err = service.checkSecurityPolicyRealizationState(&nsxGetSecurityPolicy, *(nsxGetSecurityPolicy.Path))
```

Comment:
Could you please explain why we need to check SecurityPolicy realization state?

Reply:
I checked with DFW guy, he replied: 
```
 we need a diff call to check the realization state.
hapi just returns hierarchical objects.

also , if hapi is successful - still state can be realized,unknown etc.
it doesn't guarantee realization is done
```
So, we still need a realization state check after HAPI.